ReduxSimple/ReduxStoreWithHistory.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ReduxSimple
+{
+    public abstract class ReduxStoreWithHistory<TState> : ReduxStore<TState>
+        where TState : class, new()
+    {
+        private readonly Stack<TState> _previousStates = new Stack<TState>();
```

Comment:
Would it be better to stack `Action` instead of `State`? If you have an initial state + the list of actions, you can recreate any `State` at any time.


And the main advantage is that an `Action` is generally lighter than a `State`.

Reply:
Good point, I will refactor this.

Reply:
Was thinking about it for the last 10 minutes. I think the best solution is still to only save a list of `Action`. 

But if you have executed a lot of actions and you want to go back in history, you will call the `Reduce` function too much. A smart way to handle that is to store a `State` at regular interval (of actions) in an attempt to optimize the move in history. We can plan that for a later version since it is simply an optimization and not a full feature.

Reply:
I'm thinking if we saved the action and the previous state at each step, then we can always recreate any state. Will try it out.